src/idfxt/JSONDataInterface.h

```diff
@@ -0,0 +1,102 @@
+// copyright 2014, Shannon Mackey <mackey@BUILDlab.net>
+#ifndef JSONDATAINTERFACE_H
+#define JSONDATAINTERFACE_H
+
+//for JSONDataObject and JSONDataInterface
+#include <string>
+#include <list>
+#include <map>
+#include <memory>
+
+//for old IDF text import only
+#include <algorithm>
+#include <cctype>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <chrono>
+#include <random>
+
+const std::string ACCEPTED_VERSION = "8.2";
+//end old IDF text import only
+
+extern "C"
+{
+#include "../../third_party/cJSON/cJSON.h"
+}
+
+namespace idfx {
+
+
+class JSONDataObject
+{
+public:
+    JSONDataObject(const std::string &json_content);
+    ~JSONDataObject();
+
+    std::map<std::string, std::string> getProperties();
+    std::string getPropertyValue(std::string);
+//    void setPropertyValue(std::string, std::string);
+//    void setPropertyValue(std::string, int);
+//    void setPropertyValue(std::string, double);
+
+    std::list<std::shared_ptr<JSONDataObject> > getExtensions();
```

Comment:
I find myself getting confused between extensible fields (the way we know and love them in idf), and relationships to other objects.  How do you retrieve or otherwise identify a related (referenced?) object?